### PR TITLE
Gh-3107: Refactor GafferPopGraphTest to an IT and separate util methods

### LINKED ISSUE #3107: Refactor GafferPopGraphTest to be an IT and separate util methods
GafferPopGraphTest contains tests which are ITs and not UTs, yet it is run as a UT. It should be changed to an IT.

There will be another IT added soon for the NamedOperationService. This shares common code for initialising test graphs with GafferPopGraphTest. Once this has been merged, the common code between these classes and any others should be moved to a test util class to reduce duplication and improve maintainability.

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
# Related issue

- Resolve #3107

### CODE REVIEW

**Review by t92549 — DISMISSED** [review]
Nice refactor! 👍

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphIT.java [hunk 1]
```diff
@@ -241,7 +206,7 @@ public void shouldAddAndGetVertexWithNullViewAndVertexList() {
 
         // Then
         final GafferPopVertex vertex = vertices.next();
-        assertFalse(vertices.hasNext()); // there is only 1 vertex
+        assertThat(vertices).isExhausted(); // there is only 1 vertex
```
Comment (tb06904) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphIT.java:209]: Would something like this be maybe better:
```java
assertThat(vertices).toIterable().hasSize(1);
```

Comment (tb06904) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphIT.java:209]: The `extracting()` method should allow for the last case e.g.

```java
assertThat(vertices)
                .toIterable()
                .hasSize(1)
                .first()
                .hasFieldOrPropertyWithValue("id", VERTEX_1)
                .hasFieldOrPropertyWithValue("label", SOFTWARE_NAME_GROUP)
                .extracting(v -> v.property(NAME_PROPERTY).value())
                .isEqualTo("GafferPop");
```

Comment (GCHQDeveloper314) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraphIT.java]: I've added this and included it as its own method to reduce duplication.

**Review by tb06904 — APPROVED** [review]

**Pull request merged**